### PR TITLE
Render a table widget

### PR DESCRIPTION
**Pull request by msp**
This PR is mostly just for info / discussion.

Using a stubbed backed, we are reading widget meta data, loading the data source for that widget and rendering it as a table.

The JSON interface is detailed in the /api directory. 

You can view the running app: https://dashboards-ui.firebaseapp.com/

It's probably easier to view the diffs by commit.

There's still a lot of noise in the source from the cloned starter app but we're starting to focus in on what we need and remove things we don't.

### CODE REVIEW

File: api/widgets.json [hunk 1]
```diff
@@ -5,11 +5,11 @@
           "description": "3 months of important data",
           "slug": "three-months-6stz4w",
```
Comment (nurugger07) [inline api/widgets.json]: slug is being removed from the api

**Review by nurugger07 — COMMENTED** [review]
Really just the slug removal for now

**Comment by nurugger07** [comment]
``` json
{
    "widget": {
        "name": "Three months",
        "description": "3 months of important data",
        "dataSources": [{
            "uuid": "datasource-1234",
            "name": "3 month financials"
        }],
        "adapter": "2D",
        "renderer": "TABLE",
        "refreshRate": 30,
        "insertedAt": "2017-09-04T16:03:55.948Z",
        "updatedAt": "2017-09-04T16:03:55.948Z",
        "author": {
            "username": "msp",
            "image": "https://static.productionready.io/images/smiley-cyrus.jpg"
        }
    }
}
```

The only thing I'm lacking in this ^^ is the `refreshRate`. There are several fields that I pulled out because they are not implemented and probably need to be discussed.

File: api/widgets.json (hunk 1 above)
Comment (msp) [reply api/widgets.json]: 👍

**Comment by msp** [comment]
You're right, the refreshRate is only really needed on the server if a data source is feeding a widget via a Phoenix channel..

**Pull request merged**